### PR TITLE
Don't use getattr in _get_parent_attribute to avoid recursion issues

### LINKED ISSUE #23609: include/include_role: maximum recursion depth exceeded
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
include and include_role

##### ANSIBLE VERSION
```
ansible 2.2.1.0
  config file =
  configured module search path = Default w/o overrides
```

##### CONFIGURATION
No change

##### OS / ENVIRONMENT

Red Hat Enterprise Linux Server release 6.6

##### SUMMARY
Nested include leads to the following error:
```
Exception RuntimeError: 'maximum recursion depth exceeded while calling a Python object' in <type 'exceptions.RuntimeError'> ignored
ERROR! Unexpected Exception: maximum recursion depth exceeded
```
The error occurs with real playbooks with a mix of 5 nested include_role and include
playbook:
  role
    include_role
      include_role
        include_role
          include
            include

It's reproducible with a playbook with 12 nested include.

##### STEPS TO REPRODUCE

```yaml
Playbook:
- hosts: localhost
  roles:
    - role1

role1/tasks/main.yml:
---
- name: "include: r1t1.yml"
  include: r1t1.yml

role1/tasks/r1t1.yml:
- name: "include r1t2"
  include: r1t2.yml

and so on up to role1/tasks/r1t12.yml:
---
- debug: msg="r1t12"
```

##### EXPECTED RESULTS
Playbook runs successfully.

##### ACTUAL RESULTS

```
ansible-playbook -vv test.yml
No config file found; using defaults
 [WARNING]: Host file not found: /etc/ansible/hosts

 [WARNING]: provided hosts list is empty, only localhost is available

statically included: /home/user/test/role1/tasks/r1t1.yml
statically included: /home/user/test/role1/tasks/r1t2.yml
statically included: /home/user/test/role1/tasks/r1t3.yml
statically included: /home/user/test/role1/tasks/r1t4.yml
statically included: /home/user/test/role1/tasks/r1t5.yml
statically included: /home/user/test/role1/tasks/r1t6.yml
statically included: /home/user/test/role1/tasks/r1t7.yml
statically included: /home/user/test/role1/tasks/r1t8.yml
statically included: /home/user/test/role1/tasks/r1t9.yml
statically included: /home/user/test/role1/tasks/r1t10.yml
statically included: /home/user/test/role1/tasks/r1t11.yml
statically included: /home/user/test/role1/tasks/r1t12.yml

PLAYBOOK: test.yml *************************************************************
1 plays in test.yml

PLAY [localhost] ***************************************************************
Exception RuntimeError: 'maximum recursion depth exceeded while calling a Python object' in <type 'exceptions.RuntimeError'> ignored
ERROR! Unexpected Exception: maximum recursion depth exceeded

```

[ansible-include-issue.zip](https://github.com/ansible/ansible/files/922089/ansible-include-issue.zip)

### PR DESCRIPTION
**Pull request by jimi-c**
Fixes #23609

### CODE REVIEW

**Comment by jimi-c** [comment]
!bot_skip

File: lib/ansible/playbook/block.py [hunk 1]
```diff
@@ -295,7 +295,10 @@ def _get_parent_attribute(self, attr, extend=False, prepend=False):
             if self._parent and (value is None or extend):
                 try:
                     if attr != 'when' or getattr(self._parent, 'statically_loaded', True):
-                        parent_value = getattr(self._parent, attr, None)
+                        if hasattr(self._parent, '_get_parent_attribute'):
+                            parent_value = self._parent._get_parent_attribute(attr, extend=extend, prepend=prepend)
+                        else:
+                            parent_value = self._parent._attributes.get(attr, None)
```
Comment (bcoca) [inline lib/ansible/playbook/block.py]: you need to skip value from parent if include_tasks/include_role, but still inherit

**Comment by mattclay** [comment]
CI failure in most integration tests.

**Comment by jctanner** [comment]
This terrible patch will get the integration tests to pass ...

```
]$ diff ansible.ansible/lib/ansible/playbook/task.py ansible.patched.pr/lib/ansible/playbook/task.py      
80d79                           
<     _name = FieldAttribute(isa='string', default='')           
427a427,435                     
>                               
>                         if attr == 'environment':              
>                             parent_value = self._parent._get_attr_environment()                                                  
>                         elif attr == 'tags':                   
>                             parent_value = self._parent.tags[:]                                                                  
>                         elif attr == 'no_log':                 
>                             parent_value = self._parent.no_log 
>                         elif attr == 'when':                   
>                             parent_value = getattr(self._parent, attr)
```

Hopefully that helps point out what attributes are missing from the inheritance code.

**Pull request merged**